http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/RuntimeRoutes.scala

```diff
@@ -201,13 +210,30 @@ class RuntimeRoutes(runtimeService: RuntimeService[IO], userInfoDirectives: User
       )
       _ <- runtimeService.startRuntime(userInfo, googleProject, runtimeName)
     } yield StatusCodes.Accepted
+
+  private[api] def updateRuntimeHandler(userInfo: UserInfo,
+                                        googleProject: GoogleProject,
+                                        runtimeName: RuntimeName,
+                                        req: UpdateRuntimeRequest): IO[ToResponseMarshallable] =
+    for {
+      context <- RuntimeServiceContext.generate
+      implicit0(ctx: ApplicativeAsk[IO, RuntimeServiceContext]) = ApplicativeAsk.const[IO, RuntimeServiceContext](
+        context
+      )
+      _ <- runtimeService.updateRuntime(userInfo, googleProject, runtimeName, req)
+    } yield StatusCodes.Accepted
 }
 
 object RuntimeRoutes {
-  implicit val gceConfigDecoder: Decoder[RuntimeConfigRequest.GceConfig] = Decoder.forProduct2(
-    "machineType",
-    "diskSize"
-  )((mt, ds) => RuntimeConfigRequest.GceConfig(mt, ds))
+  implicit val gceConfigDecoder: Decoder[RuntimeConfigRequest.GceConfig] = Decoder.instance { x =>
+    for {
+      machineType <- x.downField("machineType").as[Option[MachineTypeName]]
+      diskSize <- x
```

Comment:
if we change `diskSize` type to dedicated type(think there's one already?), we don't have to do the negative check here.

Reply:
Yeah that's a good idea